clarity_ext/service/dilution/service.py

```diff
@@ -632,6 +733,14 @@ def source_container_name(transfer_location):
     def target_container_name(transfer_location):
         return "END{}".format(transfer_location.container_pos)
 
+    @abc.abstractmethod
```

Comment:
If you would introduce a second variable in the settings (see the other PR) I think you could skip this abstract method.